Adds data-size attribute in g-recaptcha, with its value set in display method.

JaqueLB opened the pull request:
Issue #4 

This adds a parameter in display method call. When setted to true, the recaptcha box will be displayed with compact size. Else, the recaptcha box will be displayed with normal size, which is the default size.

JaqueLB committed:
Add the compactSize parameter in RecaptchaHelper display method.
 - The default value is false, which means that your recaptcha
will be displayed with normal size. If you set this parameter to
true, the recaptcha will be displayed with compact size.

JaqueLB committed:
Add the data-size attribute to g-recaptcha.
 - This attribute defines the size of the recaptcha box.
The attribute value is defined by $compactSize parameter,
informed when you call the RecaptchaHelper display method
inside your view.

JaqueLB committed:
Add the compact size recaptcha usage example.

anhtuank7c commented:
I don't think passing parameter through display function is  a good idea.
And please check your code standard, ensure these code passed `travis`.

How about put this configure when `loadComponent`?

ie:
```
$this->loadComponent('Recaptcha.Recaptcha', [
    'enable' => true,     // true/false
    'sitekey' => 'your_site_key', //if you don't have, get one: https://www.google.com/recaptcha/intro/index.html
    'secret' => 'your_secret',
    'type' => 'image',  // image/audio
    'theme' => 'light', // light/dark
    'lang' => 'vi',      // default en
    'size' => 'normal'  // normal | compact
]);
```

anhtuank7c commented:
We may consider `invisible` version too.

https://developers.google.com/recaptcha/docs/invisible

JaqueLB commented:
Ok, thanks. I will change the code to go in config, as soon as I can.
About the invisible recaptcha, I will look more closely in to it, probably in this weekend.

JaqueLB committed:
Remove all modifications for compact recaptcha from display.

JaqueLB committed:
Adding the size with its value to default config.

JaqueLB committed:
Adds the data-size property to recaptcha element

JaqueLB committed:
Adds the property size to test.

JaqueLB committed:
Adds the property size to README.

JaqueLB commented:
I changed the data-size code to go inside loadComponent. I have added it to tests too, and all seems ok. Please, let me know if this code is okay.

###################################################################################

Now, about the invisble recaptcha, I don't think that I have the knowledge to implement this inside a plugin at moment. 

If I understood well enough, you will need to overwrite the default submit button from Form helper to first implementation option, since the code goes inside the submit button, and a script is necessary to submit the form - check the "Automatically bind the challenge to a button" -
 https://developers.google.com/recaptcha/docs/invisible#auto_render

The second implementation option seems to be similar to the Explicity Implementation of normal recaptcha, but I think that isn't the implementation used in your plugin.

The invisible recaptcha does not have the same sitekey from Google, so I don't know if it will be possible to implement this in same plugin, maybe you will need to implement an option inside loadComponent for choose between the normal or the invisible recaptcha, and then this would be used to decide which Element to use in display, because each implementation is different.
